Added more regional flavor jobs

Amazinite opened the pull request:
This PR essentially finishes the work started by @Pointedstick in #1854, giving each region of space their own flavor jobs.

Here are the jobs added by this PR, along with little descriptions about them:
* Dirt Belt
  * **Drought relief (cargo)**: The breadbasket of the Republic experiences droughts from time to time.
  * **Harvest convoy (escort)**: All that food has to get to the population centers somehow.
  * **Remembrance Day celebration (passenger/cargo)**: The Dirt Belt holds the biggest celebrations for Remembrance Day every year.
* Frontier
  * **Disaster relief (cargo)**: Settlements on Frontier worlds aren't in the safest of places, and not every world has the infrastructure to handle disasters.
  * **Blockade running (cargo)**: Frontier worlds are still developing, which means they need shipments of supplies delivered to their dangerous regions of space.
  * **Emigration (passenger)**: Frontier worlds aren't the best places to be, so it's only reasonable that some people want to move elsewhere when given the chance.
* Near Earth
  * **Earth Day celebration (passenger/cargo)**: Why wouldn't Earth Day become a prevalent holiday in an era where humanity spans the stars? 
  * **Historical Trip (passenger)**: Kids from wealthy schools get to go on field trips to historical locations near Earth. The same can't be said for kids in the Dirt Belt though.
* Rim
  * **Han Sizer month (passenger)**: Every year, the Rim celebrates the man who first explored the Rim, Han Sizer. He took an entire month to explore the Rim and all its planets, leading to the celebration lasting as long as it does.
  * **Southbound and Kraz shipments (cargo)**: The high-piracy region of the Rim relies on Southbound and Kraz to develop sturdy ships and powerful weaponry respectively. 
* South
  * **Pirate occupations (combat)**: Just read the spaceport description of Hans and you'll understand. Now a spaceport mission and not a job.
  * **Sketchy cargo/passenger**: Trinket isn't the only place that pirates in disguise hang out.

Edit: I also renamed `syndicate world jobs.txt` to just `syndicate jobs.txt` since it didn't fit with other regional job file names.

Amazinite commented:
To note: the blockade running frontier jobs are not intended to fulfill the blockade running missions task mentioned in the [blog post](https://endless-sky.github.io/2016/11/21/easy-ways-to-get-involved.html) almost a year ago now. I intend to create actual blockade running missions (not just jobs) that have more story to them and are tailored to the three frontier regions (north, core, and south).

Pointedstick commented on data/frontier jobs.txt:
```diff
@@ -0,0 +1,174 @@
+	name "Frontier delivery to <planet>"
+	job
+	repeat
+	description `Deliver <cargo> to the frontier world of <destination>. Frontier worlds have a high rate of piracy, so a pirate prescense is expected. Payment is <payment>.`
```
prescense -> presence

Pointedstick commented on data/dirt belt jobs.txt:
```diff
@@ -0,0 +1,279 @@
+			variant
+				"Behemoth" 2
+	on complete
+		payment 220000
```
That's a lot of money, even for large pirate fleets.

Amazinite replied:
The distance that needs to be traveled can also be over double that of normal escort jobs, but if you want me to toughen up the pirates I can.

Pointedstick commented on data/frontier jobs.txt:
```diff
@@ -0,0 +1,174 @@
+	name "Frontier delivery to <planet>"
+	job
+	repeat
+	description `Deliver <cargo> to the frontier world of <destination>. Frontier worlds have a high rate of piracy, so a pirate prescense is expected. Payment is <payment>.`
```
prescense -> presence

Pointedstick commented on data/frontier jobs.txt:
```diff
@@ -0,0 +1,174 @@
+	name "Frontier delivery to <planet>"
+	job
+	repeat
+	description `Deliver <cargo> to the frontier world of <destination>. Frontier worlds have a high rate of piracy, so a pirate prescense is expected. Payment is <payment>.`
```
prescense -> presence

Pointedstick commented on data/near earth jobs.txt:
```diff
@@ -0,0 +1,172 @@
+		attribute "paradise"
+	on complete
+		payment 5000 250
+		dialog `A security detail on <planet> escort the Parliament members from your ship, and you collect your payment of <payment>.`
```
escort -> escorts

"security detail" is singular, even though it's a collective noun. Alternatively, this could be rewritten as "The members/guards/soldiers/etc. of a security detail escort..."

Bladewood replied:
The problem we had with Deep Security.

Pointedstick commented on data/dirt belt jobs.txt:
```diff
@@ -0,0 +1,279 @@
+# Copyright (c) 2017 by Michael Zahniser
```
For these new files you're adding, probably the copyright is yours, or at least in addition to Michael's.

Amazinite replied:
I created the file, but any number of changes could be added in the future such that the work I originally added is only a minority of the file, so I wouldn't really feel justified in attaching my name to it. The fact that I made the file is recorded on Github anyway.

Pointedstick commented on data/rim jobs.txt:
```diff
@@ -0,0 +1,144 @@
+	name `Han Sizer Celebration`
+	job
+	repeat
+	description `Han Sizer month commemorates the first explorer of the Rim. To celebrate, citizens visit as many planets in the Rim as they can in the month. Bring <fare> to all the marked systems and return to <destination> for <payment>.`
```
I love this mission; so creative!

Bladewood replied:
It might be better to just throw the complete information on the nature of Han Sizer Month into some planet description and only mention in the job descriptions that people are moving around to celebrate it, in the same style as the Coalition Games jobs.

Also, you might want to decapitalize "celebration". And/or capitalize the first "month".

Amazinite replied:
I could possibly do that. Longjump, the first planet in the Rim, has an abysmally tiny spaceport and planet description that could use some filling.

Amazinite replied:
I've added information about Han Sizer month to the description of Longjump.

Pointedstick commented on data/south jobs.txt:
```diff
@@ -0,0 +1,158 @@
+		fleet "Small Southern Pirates" 3
+	on complete
+		payment 100000
+		dialog `The government of <planet> thank you for defeating the pirate and pay you <payment>.`
```
pirate -> pirates

Bladewood replied:
Also, it should be "thanks", since government is singular.

Pointedstick commented on data/south jobs.txt:
```diff
@@ -0,0 +1,158 @@
+		fleet "Large Southern Pirates"
+	on complete
+		payment 150000
+		dialog `The government of <planet> thank you for defeating the pirate and pay you <payment>.`
```
pirate -> pirates

Pointedstick commented on data/south jobs.txt:
```diff
@@ -0,0 +1,158 @@
+		fleet "Large Southern Pirates" 3
+	on complete
+		payment 200000
+		dialog `The government of <planet> thank you for defeating the pirate and pay you <payment>.`
```
pirate -> pirates

Pointedstick commented on data/south jobs.txt:
```diff
@@ -0,0 +1,158 @@
+		distance 2 10
+		attributes "south"
+	on accept
+		dialog `You follow the directions specified of the job to an abandoned-looking warehouse and find your cargo in unlabeled crates. As you load it on to your ship, you get the feeling that you are being watched.`
```
on to -> onto

Pointedstick commented on data/south jobs.txt:
```diff
@@ -0,0 +1,158 @@
+		distance 2 10
+		attributes "south"
+	on accept
+		dialog `You follow the directions specified of the job to an abandoned-looking warehouse and find your cargo in unlabeled crates. As you load it on to your ship, you get the feeling that you are being watched.`
```
on to -> onto

Pointedstick commented:
Excellent work. I really need to finish my own versions of these. I'll be more than happy to add them once this is merged (as I have no doubt it will be, given the quality and attention to detail).

In general, some of the random conditions might need to be adjusted down a bit to prevent the jobs being too frequent. A lot of the flavor comes from the relative rarity IMHO.

DrBlight commented on data/south jobs.txt:
```diff
@@ -0,0 +1,158 @@
+		dialog `You follow the directions specified of the job to an abandoned-looking warehouse and find your cargo in unlabeled crates. As you load it on to your ship, you get the feeling that you are being watched.`
+	on complete
+		payment 0 160
+		dialog "You drop off the unlabeled cargo in a warehouse not too disimilar to the one you picked it up in. A man dressed in a cloak hands you <payment> after the last crate is removed from your ship."
```
I'm afraid it's spelled "dissimilar"

DrBlight replied:
I'm afraid it's spelled "dissimilar"

DrBlight commented on data/south jobs.txt:
```diff
@@ -0,0 +1,158 @@
+		dialog `You follow the directions specified of the job to an abandoned-looking warehouse and find your cargo in unlabeled crates. As you load it on to your ship, you get the feeling that you are being watched.`
+	on complete
+		payment 0 160
+		dialog "You drop off the unlabeled cargo in a warehouse not too disimilar to the one you picked it up in. A man dressed in a cloak hands you <payment> after the last crate is removed from your ship."
```
"dissimilar"

DrBlight replied:
"dissimilar"

DrBlight commented:
I've been using this addition and I think it's great! I love the missions that add a little more flavor.

Bladewood commented on data/dirt belt jobs.txt:
```diff
@@ -0,0 +1,279 @@
+	name `Food convoy to <planet>`
+	job
+	repeat
+	description `The captain of the <npc> will pay you <payment> to escort his convoy of ships carrying food to <destination> by <date>.`
```
It's not necessarily something I agree with, but I believe "his" should be changed to "their", for the sake of gender ambiguity.

Pointedstick replied:
Yes, and in fact it should be rewritten to make the gender ambiguity not be awkward. For example:

`The captain of the <npc>, leading a convoy of ships carrying food, will pay you <payment> to escort them to <destination> by <date>.`

Or something like that.

Bladewood commented on data/frontier jobs.txt:
```diff
@@ -0,0 +1,174 @@
+	name "Emigration to <planet>"
+	job
+	repeat
+	description "These <bunks> passengers have finally saved up enough money to move to a more prosperous world. Bring them to <destination> where they will start their new life. Payment is <payment>."
```
It might be better to rewrite this into something more concise:
`These <bunks> passengers have finally saved up enough money to start a new life on <destination>, and are willing to pay you <payment> to transport them.`

Maybe the `up` in `saved up enough` can be taken out.

Bladewood commented on data/near earth jobs.txt:
```diff
@@ -0,0 +1,172 @@
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+mission "Historical trip to <planet>"
```
I think it'd be better to emphasize the field trip part, not the historical trip one.

Amazinite replied:
For what reason? The reason I emphasize the fact that it's a historical trip is because it's to the near Earth worlds, the oldest (known) colonies in human space, and even the planets of Sol, tying it more to the fact that it's a near Earth job instead of making it more general, in which case it would sound like it can offer anywhere.

Bladewood replied:
It sounds like someone's making the trip of his life that would eventually be recognized as a historical event, a paragraph for which is then placed in every history textbook.
That is, it sounds like it's about the actual Pilgrims coming to America, not a bunch of schoolchildren a thousand years into the future going on a field trip to explore their achievements.

Bladewood commented on data/rim jobs.txt:
```diff
@@ -0,0 +1,144 @@
+		attributes "rim"
+	on complete
+		payment 15000 170
+		dialog `Your passengers thank you for taking them on a journey around the Rim, and you collect you payment of <payment>.`
```
`and you collect you payment of <payment.`

DrBlight commented:
So I'm being offered an Earth Day mission...while presently docked on Earth. Whoops!
![earthdayonearth](https://user-images.githubusercontent.com/33576732/32696040-29213370-c72a-11e7-9213-eccfc695f400.png)

Amazinite commented:
Oops. The cargo Earth Day job doesn't have a source distance from Sol.

Pointedstick commented on data/dirt belt jobs.txt:
```diff
@@ -0,0 +1,279 @@
+	job
+	repeat
+	description `The farming world of <destination> is currently experiencing a drought. Deliver <cargo> to the planet by <date> for <payment>.`
+	cargo "drought relief" 25 2 .05
```
drought relief -> drought relief supplies

Or maybe "water"?

Brick63 replied:
well it could be medication, food, or any other number of things that happen due to a lack of water

Pointedstick commented on data/frontier jobs.txt:
```diff
@@ -0,0 +1,174 @@
+	job
+	repeat
+	description "A disaster has occured in a colony on the frontier world of <destination>. Deliver <cargo> to the planet by <date> for <payment>."
+	cargo "disaster relief" 25 2 .05
```
"disaster relief supplies"

tehhowch commented on data/dirt belt jobs.txt:
```diff
@@ -0,0 +1,279 @@
+	description `Bring <fare> home to <destination> after the Remembrance Day celebration. Payment is <payment>.`
+	passengers 4 4 .5
+	to offer
+		random < 60
```
malformed ConditionSet here, needs an `or` below.
Seems kinda high offer rate too, considering it offers for up to 8 weeks. It may be better to rewrite this so that the offer rate decreases as time goes on. You'd probably need features from #3128 to do that smoothly, but could use more randoms to be a bit more piecewise about it.

tehhowch commented on data/dirt belt jobs.txt:
```diff
@@ -0,0 +1,279 @@
+	description `Bring <fare> home to <destination> after the Remembrance Day celebration. Payment is <payment>.`
+	passengers 4 4 .3
+	to offer
+		random < 60
```
see above

tehhowch commented on data/near earth jobs.txt:
```diff
@@ -0,0 +1,174 @@
+		random < 80
+		or
+			and
+				month == 2
```
You could put additional `random` checks in these `and` nodes to roughly gate the offer chance.. e.g. 20% offer in late feb, 40% in March, and the full 80% in April.

tehhowch commented on data/near earth jobs.txt:
```diff
@@ -0,0 +1,174 @@
+		random < 70
+		or
+			and
+				month == 4
```
As above, throwing in extra `random` checks would let you phase this job out by month.

tehhowch commented on data/rim jobs.txt:
```diff
@@ -0,0 +1,144 @@
+	description "Deliver <cargo> to <destination>, the home of Southbound Shipyards, the largest ship manufacturer in the region. Payment is <payment>."
+	cargo "weapons" 5 2 .1
+	to offer
+		random < 50
```
These will stack up real quick at 50%, and with them all going to the same location, makes for nice job strategy.

Pointedstick replied:
Indeed; these should probably have lower frequencies.

tehhowch commented on data/rim jobs.txt:
```diff
@@ -0,0 +1,144 @@
+	description "Deliver <cargo> to <destination>, the home of Kraz Cybernetics, the largest weapons manufacturer in the region. Payment is <payment>."
+	cargo "weapons parts" 5 2 .1
+	to offer
+		random < 50
```
same as above

tehhowch commented on data/rim jobs.txt:
```diff
@@ -0,0 +1,144 @@
+	source
+		near "Zug" 2 10
+		attributes "rim" "south"
+	destination "Zug"
```
Should not be Zug if this is going to Kraz

tehhowch commented on data/south jobs.txt:
```diff
@@ -0,0 +1,158 @@
+		distance 2 10
+		attributes "south"
+	on accept
+		dialog `You follow the directions specified of the job to an abandoned-looking warehouse and find your cargo in unlabeled crates. As you load it onto your ship, you get the feeling that you are being watched.`
```
Needs rephrasing

Amazinite commented on data/dirt belt jobs.txt:
```diff
@@ -0,0 +1,279 @@
+		attributes "dirt belt"
+	on complete
+		payment 1000 180
+		dialog "You wish your <passengers> the best of luck on <planet>. The children run off your ship in green face paint, and the parents pay you <payment>."
```
>The trip is very likely completed before the actual Remembrance Day - should the children really be painted up, even if it's still not even the first of the coming year?

Go talk to the people who put Christmas decorations up the day after Halloween. 
That's my explanation for it. :)
It could also be that the celebration for Remembrance Day starts earlier than the day itself, but the biggest party is on the actual day.

tehhowch replied:
As long as the existing lore is matched, all is well.

>In these days when the Navy ensures peace throughout much of the galaxy, on many worlds the Alphas are treated as nothing more than a bogeyman story to frighten children. Because the most striking physical feature of the Alphas was their photosynthetic skin, on some worlds kids even run around wearing green face paint on Remembrance Day, trying to frighten people. But on other worlds, especially on the fringes of space where the Navy seldom patrols, the Alphas are still spoken of with fear.

endless-sky replied:
I'm not sure it makes sense for Remembrance Day to involve travel between planets. And, it especially doesn't make sense that Dirt Belt inhabitants would spend money on that sort of travel.

Amazinite replied:
I imagine that some people may travel for the purpose of seeing family (I'm sure there are *some* families that are spread across multiple planets), or perhaps just going to a planet that has a bigger celebration that they would like to partake in. Would it be acceptable if I kept the jobs but significantly reduced the offer rate as a representation of how only a small number of regional inhabitants are able to travel?

Amazinite commented:
From celebrations now taper off 2 months after the celebration ends.
To celebration missions now start off with a medium random condition, spike higher the entire month before the celebration, then lower as the celebration approaches.

Pointedstick commented:
I've been finding these offered on Pirate planets! Perhaps add `not attributes Pirate`

Pointedstick commented on data/near earth jobs.txt:
```diff
@@ -0,0 +1,185 @@
+	repeat
+	description `Bring these <bunks> Parliament members to their vacation retreat on <destination>.`
+	passengers 2 10 .9
+	deadline
```
This job has a deadline, but the description doesn't list it.

Amazinite replied:
It doesn't list the payment either. :)

Pointedstick commented on data/near earth jobs.txt:
```diff
@@ -0,0 +1,185 @@
+	repeat
+	description `Bring <bunks> members of an election campaign to <destination> by <date> for a political event. The campaign will pay you <payment>.`
+	passengers 5 5 .2
+	deadline
```
This job has a deadline, but the description doesn't list it.

Amazinite commented:
@Pointedstick It looks like a few frontier jobs had sources (or destinations) which allowed it to be a pirate planet; remedied this by including a government filter.

Amazinite commented on data/dirt belt jobs.txt:
```diff
@@ -0,0 +1,279 @@
+	to offer
+		random < 10
+	source
+		government "Republic"
```
Done

Pointedstick commented on data/near earth jobs.txt:
```diff
@@ -0,0 +1,185 @@
+	source "Earth"
+	destination
+		distance 3 10
+		attribute "paradise"
```
attribute -> attributes

endless-sky commented:
My main concern is that in some ways, having a smaller number of distinct job types is preferable because the player can get a sense of what jobs are generally offered (run, cargo, large rush, large cargo) and know what to expect in terms of payoff and deadline for a job of that type. Adding lots of different job types makes it harder to read through the jobs list if all you're really looking for is basic money-making missions.

Because of that and the issue that any job board mission needs to have very bland text so it doesn't seem weird for it to keep repeating, jobs with a lot of "flavor" might be better off as spaceport missions rather than job board missions.

Amazinite commented:
>jobs with a lot of "flavor" might be better off as spaceport missions rather than job board missions.

Which ones would those be? 
I'm thinking of the three sets of celebration missions + the election missions as having the most "flavor" to them, and probably the southern pirate occupation missions as well, but I'm not exactly sure which ones you're referring to.

Amazinite commented:
I've switched the pirate occupation jobs to be spaceport missions, removed the Parliament vacation job and the election year job (meaning that this PR is now 100% compatible with the vanilla game given that a modulus operator is no longer needed), and drastically reduced the chances that a family will be traveling in the Dirt Belt for Remembrance day while also increasing the chance of moving celebration supplies by adding a second job.

Are there any other existing jobs...
>with a lot of "flavor" [that] might be better off as spaceport missions rather than job board missions

?

Pointedstick commented:
I'd really love for this to get merged. These kind of jobs just add flavor and variety; not everything needs to be a special one-time-only missions triggered by visiting the spaceport.

Fzzr commented on data/dirt belt jobs.txt:
```diff
@@ -0,0 +1,325 @@
+		system destination
+		fleet "Small Northern Pirates"
+	npc
+		government Pirate
```
Needs quotes around `Pirate`

tehhowch commented on data/south jobs.txt:
```diff
@@ -0,0 +1,213 @@
+		fleet "Small Southern Pirates" 3
+	on offer
+		conversation
+			`Suddenly you hear raised voices and shouting outside as pilots run for their ships: "<destination> is under attack by a small pirate fleet! They are requesting help from any combat-worthy ship in the region to join in the defenses!" The authorities of <planet> will probably pay you quite well if you assist them, but this could also be an easy way to get yourself killed.`
```
#3468 at minimum

Also, consider defining a stock `conversation` (see #3384). I note there are slight differences (small / ___ / large) but perhaps you are fine with dropping that.

Is it possible for the destination planets here to actually be very capable of defending themselves (e.g. a Free Worlds naval base?) `attributes south`, 2-3 jumps from `south` or `rim` doesn't seem very specific.

Amazinite replied:
Part of the whole Southern Defense Pact/FW deal is that worlds defend one another. The Rim is less susceptible to piracy due to how narrow the entrance is, but southern worlds are all rather close to the pirate planets of the south. A planet may be able to defend itself, but you wouldn't hear about it then, now would you. :)

tehhowch commented on data/south jobs.txt:
```diff
@@ -0,0 +1,201 @@
+
+conversation "pirate occupation"
+	`Suddenly you hear raised voices and shouting outside as pilots run for their ships: "<destination> is being occupied by a pirate fleet! They are requesting help from any combat-worthy ship in the region to join the defenses!" The authorities of <planet> will probably pay you quite well if you assist them, but this could also be an easy way to get yourself killed.`
+		choice
```
I think this is indented one too far. Running the parse test throws this back:
```
Expected goto or endpoint in conversation, found this:
file "/home/tehhowch/projects/endless-sky/data/south jobs.txt"
  conversation "pirate occupation"
    `Suddenly you hear raised voices and shouting outside as pilots run for their ships: "<destination> is being occupied by a pirate fleet! They are requesting help from any combat-worthy ship in the region to join the defenses!" The authorities of <planet> will probably pay you quite well if you assist them, but this could also be an easy way to get yourself killed.`
      choice

Expected goto or endpoint in conversation, found this:
file "/home/tehhowch/projects/endless-sky/data/south jobs.txt"
  conversation "pirate occupation"
    `Suddenly you hear raised voices and shouting outside as pilots run for their ships: "<destination> is being occupied by a pirate fleet! They are requesting help from any combat-worthy ship in the region to join the defenses!" The authorities of <planet> will probably pay you quite well if you assist them, but this could also be an easy way to get yourself killed.`
      "	(Stay here until the fight is over.)"

Expected goto or endpoint in conversation, found this:
file "/home/tehhowch/projects/endless-sky/data/south jobs.txt"
  conversation "pirate occupation"
    `Suddenly you hear raised voices and shouting outside as pilots run for their ships: "<destination> is being occupied by a pirate fleet! They are requesting help from any combat-worthy ship in the region to join the defenses!" The authorities of <planet> will probably pay you quite well if you assist them, but this could also be an easy way to get yourself killed.`
      "	(Join the defense fleet.)"
```

tehhowch replied:
Probably the `launch` is too far indented as well. Parse test doesn't pick up on over-indentation at the moment.

jafdy replied:
Yep, both `choice` and `launch` are overindented.

Amazinite replied:
Were*

Pointedstick commented on data/frontier jobs.txt:
```diff
@@ -0,0 +1,181 @@
+	name "Frontier delivery to <planet>"
+	job
+	repeat
+	description `Deliver <cargo> to the frontier world of <destination>. Frontier worlds have a high rate of piracy, so a pirate presence is expected. Payment is <payment>.`
```
[for all of these jobs]

I don't like "pirate" and "piracy" in close proximity. Maybe change this sentence to "Frontier worlds have a high rate of piracy, so be prepared for attack." (or something like that)

Pointedstick commented:
Perfecto :ok_hand:
